slides/kube/concepts-k8s.md

```diff
@@ -98,39 +98,76 @@ class: pic
 
 ---
 
-## Kubernetes architecture: the master
+## Kubernetes architecture: the nodes
+
+- The nodes executing our containers run a collection of services:
+
+  - a container Engine (typically Docker)
+
+  - kubelet (the "node agent")
+
+  - kube-proxy (a necessary but not sufficient network component)
+
+- Nodes were formerly called "minions"
+
+  (You might see that word in older articles or documentation)
+
+---
+
+## Kubernetes architecture: the control plane
 
 - The Kubernetes logic (its "brains") is a collection of services:
 
   - the API server (our point of entry to everything!)
+
   - core services like the scheduler and controller manager
+
   - `etcd` (a highly available key/value store; the "database" of Kubernetes)
 
-- Together, these services form what is called the "master"
+- Together, these services form the control plane of our cluster
 
-- These services can run straight on a host, or in containers
-  <br/>
-  (that's an implementation detail)
+- The control plane is also called the "master"
 
-- `etcd` can be run on separate machines (first schema) or co-located (second schema)
+---
+
+## Running the control plane on special nodes
+
+- It is common to reserve a dedicated node for the control plane
+
+  (Except for single-node development clusters, like when using minikube)
+
+- This node is then called a "master"
+
+  (Yes, this is ambiguous: is the "master" a node, or the whole control plane?)
+
+- Normal applications are restricted from running on this node
+
+  (By using a mechanism called ["taints"](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/))
+
+- When high availability is required, each service of the control plane must be resilient
 
-- We need at least one master, but we can have more (for high availability)
+- The control plane is then replicated on multiple nodes
+
+  (This is sometimes called a "multi-master" setup)
 
 ---
 
-## Kubernetes architecture: the nodes
+## Running the control plane outside containers
 
-- The nodes executing our containers run another collection of services:
+- The services of the control plane can run in or out of containers
 
-  - a container Engine (typically Docker)
-  - kubelet (the "node agent")
-  - kube-proxy (a necessary but not sufficient network component)
+- For instance: since `etcd` is a critical service, some people
+  deploy it directly on a dedicated cluster (without containers)
 
-- Nodes were formerly called "minions"
+  (This is illustrated on the first "super complicated" schema)
+
+- In some hosted Kubernetes offerings (e.g. GKE), the control plane is invisible
```

Comment:
Do we want to name the three major cloud providers' hosted offerings here? (I am actually not sure if "invisible" is accurate for AWS, but I can imagine people might ask. Also "invisible" is ambiguous. By "see", do you mean "we only interact with"?

Reply:
I only listed GKE because I don't know what the control plane "looks like" on EKS and AKS...

I'll be happy to clarify when I have the time to investigate more (I just need to stage up a cluster on each of them! This could actually be a super interesting chapter)